language/go/generate.go

```diff
@@ -41,7 +41,8 @@ func (gl *goLang) GenerateRules(args language.GenerateArgs) language.GenerateRes
 	protoPackages := make(map[string]proto.Package)
 	protoFileInfo := make(map[string]proto.FileInfo)
 	for _, r := range args.OtherGen {
-		if r.Kind() != "proto_library" {
+		// TODO: Apply kind mapping here (and similar tests below)?
```

Comment:
I think this change should be made in a way that doesn't require any collaboration from the language extensions. Go shouldn't need to know the mapped kind for `proto_library`, and a bug in Go shouldn't cause it to emit `go_binary` instead of the mapped kind.

Instead, maybe something in fix-update.go could call `Rule.SetKind` on the result of `GenerateRules`, before the rules are merged into files? That would allow language extensions to generate rules normally, and they would see the unmapped rules generated by other extensions.

Reply:
Great point, very much agree with that. Will do.